crates/plex-api/src/server/transcode.rs

```diff
@@ -1023,3 +1085,368 @@ where
+    client: HttpClient,
+    state: QueueItemState,
+}
+
+impl fmt::Debug for QueueItem {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.state.fmt(f)
+    }
+}
+
+impl QueueItem {
+    pub fn id(&self) -> u32 {
+        self.state.id
+    }
+
+    pub fn key(&self) -> &str {
+        &self.state.key
+    }
+
+    pub fn status(&self) -> QueueItemStatus {
+        self.state.status.clone()
+    }
+
+    /// If this item is currently being transcoded this will return the current
+    /// transcode stats.
+    pub fn stats(&self) -> Option<TranscodeSessionStats> {
+        self.state.session_stats.clone()
+    }
+
+    /// If the status is an error this may reveal more information.
+    pub fn error(&self) -> Option<&str> {
+        self.state.error.as_deref()
+    }
+
+    /// Returns true if this item was or is being transcoded. If false then
+    /// downloading will just download the original media file.
+    pub fn is_transcode(&self) -> bool {
+        self.state.decision_result.direct_play_decision_code != Some(1000)
+    }
+
+    /// Returns the container format of the file that will be downloaded.
+    ///
+    /// This will fail if the item is not available.
+    pub async fn container(&self) -> Result<Option<ContainerFormat>> {
+        // The API doesn't appear to expose the container format in a
+        // a particularly nice way. If the item is in the middle of transcoding
+        // then it is available in the transcode stats, but if already complete
+        // the stats are no longer exposed. However the content-disposition
+        // header of the download endpoint does include the filename complete
+        // with correct extension for the container so we can use that.
+
+        let path = DOWNLOAD_QUEUE_DOWNLOAD
+            .replace("{queueId}", &self.state.queue_id.to_string())
+            .replace("{itemId}", &self.state.id.to_string());
+
+        let response = self.client.head(path).send().await?;
+        match response.status() {
+            StatusCode::OK => {
+                if let Some(val) = response.headers().get(CONTENT_DISPOSITION) {
+                    if let Ok(st) = val.to_str() {
+                        if let Some((_, Some(ext))) = parse_content_disposition(st).filename() {
+                            Ok(ContainerFormat::from_str(&ext).ok())
+                        } else {
+                            Ok(None)
+                        }
+                    } else {
+                        Ok(None)
+                    }
+                } else {
+                    Ok(None)
+                }
+            }
+            StatusCode::SERVICE_UNAVAILABLE => Err(Error::TranscodeIncomplete),
+            _ => Err(crate::Error::from_response(response).await),
+        }
+    }
+
+    /// Returns the expected length of the download.
+    ///
+    /// This will fail if the item is not available.
+    pub async fn len(&self) -> Result<Option<u64>> {
```

Comment:
I have some conflicting feelings about the naming here. What are your thoughts about renaming it to `content_length()` or something along those lines?